api/src/controllers/schedule.controller.ts

```diff
@@ -0,0 +1,89 @@
+import { ISchedule, Days } from '../interfaces/Schedule';
+import { create_schedule } from '../helpers/createMockData';
+import { IUser } from '../interfaces/User';
+
+const Schedule = require('../models/schedule.model');
+const User = require('../models/user.model');
+const jwt_decode = require('jwt-decode');
+
+/**
+ * Schedule Controller
+ *
+ * @export
+ * @class ScheduleController
+ */
+export class ScheduleController 
```

Comment:
Also moving forward, lets write tests for the files we are creating. I understand these are subject to change. But if we have tests written and are confirming object types and data. THen updating the response or input will break tests but is easy to update.

Reply:
Added tests for schedule controller and teacherpref controller.